Tasks/AzureFunctionAppV2/operations/AzureAppServiceUtility.ts

```diff
@@ -122,9 +122,18 @@ export class AzureAppServiceUtility {
 
     public async getKuduService(): Promise<Kudu> {
         var publishingCredentials = await this._appService.getPublishingCredentials();
+        var scmPolicyCheck = await this.isSitePublishingCredentialsEnabled();            
```

Comment:
Should we fallback to basic auth (old path) if this call fails for some reason ? would that be more safe from compat perspective ? Or do you think SPN always have access to get this info ?

Could there be a case where SPN permissions are modified since it was created ? i.e it was able to access publish profile during pipeline setup but not able to anymore during pipeline run.

Reply:
I will test again tomorrow but I had an error in the ARM call which failed and it still used Basic Authentication.

Reply:
Can you please educate me how we will use Basic Auth when ARM call fails ? 

If ARM call fails , **isSitePublishingCredentialsEnabled** return **false**. No ? Then the code here tries to do Auth using Bearer token ?

Reply:
Hi, I changed the logic so that we don't fall back to Basic Auth if the token call fails. The logic is if property is set to false use bearer, if it is true use Basic Auth other wise use Bearer token.